File: A/Memory.java
public class Memory extends Hardware{
    private int freq;
    private int memSize;
    private boolean supportsCuda;

    public Memory(){
        this.freq = 0;
        this.memSize = 0;
        this.supportsCuda = false;
    }

    public void setFreq(int f){
        this.freq = f;
    }

    public int getFreq(){
        return freq;
    }

    public void setMemSize(int m){
        this.memSize = m;
    }

    public int getMemSize(){
        return memSize;
    }

    public void setSupportsCuda(boolean cs){
        this.supportsCuda = cs;
    }

    public boolean getSupportsCuda(){
        return supportsCuda;
    }
}

File: B/Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
    __freq = 0
    __memSize = 0
    __supportsCuda = False
    
    def __init__(self):
        self.__freq = 0
        self.__memSize = 0
        self.__supportsCuda = False
    
    def setFreq(self, f):
        self.__freq = f
    
    def getFreq(self):
        return self.__freq
    
    def setMemSize(self, m):
        self.__memSize = m
    
    def getMemSize(self):
        return self.__memSize

    def setCuda(self, c):
        self.__supportsCuda = c
    
    def getCuda(self):
        return self.__supportsCuda